src/Function/Bundles/Related.agda

```diff
@@ -0,0 +1,290 @@
+------------------------------------------------------------------------
+
+-- The Agda standard library
+--
+-- Relatedness for the function hierarchy
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Function.Bundles.Related where
+
+open import Level
+open import Relation.Binary using (Sym; Reflexive; Trans; IsEquivalence; Setoid; IsPreorder; Preorder)
+import Function.Related as R
+open import Function.Bundles
+open import Function.Base
+open import Function.Equality using (_⟨$⟩_)
+import Function.Equivalence as Eq
+import Function.Injection   as Inj
+import Function.Inverse     as Inv
+import Function.LeftInverse as LeftInv
+import Function.Surjection  as Surj
+import Relation.Binary.PropositionalEquality as P
+
+open import Function.Properties.Inverse
+import Function.Construct.Symmetry as Symmetry
+import Function.Construct.Identity as Identity
+import Function.Construct.Composition as Composition
+
+open import Data.Product
+
+infix 4 _∼[_]_
+
+_∼[_]_ : ∀ {ℓ₁ ℓ₂} → Set ℓ₁ → R.Kind → Set ℓ₂ → Set _
+A ∼[ R.implication         ] B = A ⟶ B
+A ∼[ R.reverse-implication ] B = B ⟶ A
+A ∼[ R.equivalence         ] B = A ⇔ B
+A ∼[ R.injection           ] B = A ↣ B
+A ∼[ R.reverse-injection   ] B = B ↣ A
+A ∼[ R.left-inverse        ] B = A ↪ B
+A ∼[ R.surjection          ] B = A ↠ B
+A ∼[ R.bijection           ] B = A ⤖ B
+
+-- A non-infix synonym.
+
+Related : R.Kind → ∀ {ℓ₁ ℓ₂} → Set ℓ₁ → Set ℓ₂ → Set _
+Related k A B = A ∼[ k ] B
+
+module _ {ℓ₁ ℓ₂} {A : Set ℓ₁} {B : Set ℓ₂} where
+
+  toRelated : {K : R.Kind} → A ∼[ K ] B → A R.∼[ K ] B
```

Comment:
I think these conversions should live in the soon-to-be deprecated file `Function.Related`.

Reply:
I'll move this as well!